tests/test_contributor_read_permissions.py

```
@@ -0,0 +1,254 @@
+
+    def test_unauthenticated_denied(self):
+        perm = IsAdminOrModeratorOrContributorReadOnly()
+        req = _build_request("GET")
+        req.user = None
+        assert not perm.has_permission(req, _MockView())
+
+    def test_unauthorized_user_denied(self):
+        user = create_user_with_role("norole", "norole@example.com", "NoRole")
+        # NoRole is not a real group; create a plain user
+        plain = User.objects.create_user(
+            username="plain", email="plain@example.com", password="testpass123"
```

Ruff (0.15.15)

[error] Possible hardcoded password assigned to argument: "password"

(S106)